spec/graphql/schema/subscription_spec.rb

```diff
@@ -437,4 +444,59 @@ def in_memory_subscription_count
       assert_equal 1, in_memory_subscription_count
     end
   end
+
+  describe "`subscription_scope` method" do
+    it "provdes a subscription scope that is recognized in the schema" do
+      scoped_subscription = SubscriptionFieldSchema::get_field("Subscription", "directTootWasTooted")
+  
+      assert_equal :viewer, scoped_subscription.subscription_scope
+    end
+  
+    it "provides a subscription scope that is used in execution" do
+      res = exec_query <<-GRAPHQL, context: { viewer: :me }
+        subscription {
+          directTootWasTooted {
+            toot { body }
+          }
+        }
+      GRAPHQL
+      assert_equal 1, in_memory_subscription_count
+
+      # Only the subscription with scope :me should be in the mailbox
+      obj = OpenStruct.new(toot: { body: "Hello from matz!" }, user: SubscriptionFieldSchema::USERS["matz"])
+      SubscriptionFieldSchema.subscriptions.trigger(:direct_toot_was_tooted, {}, obj, scope: :me)
+      SubscriptionFieldSchema.subscriptions.trigger(:direct_toot_was_tooted, {}, obj, scope: :not_me)
+      mailbox = res.context[:subscription_mailbox]
+
+      assert_equal 1, mailbox.length
+
+      expected_response = {
+        "data" => {
+          "directTootWasTooted" => {
+            "toot" => {
+              "body" => "Hello from matz!"
+            }
+          }
+        }
+      }
+
+      assert_equal expected_response, mailbox.first
+    end
+
+    it "allows for proper inheritance of the class's configuration through `field_options`" do
+      direct_toot_options = SubscriptionFieldSchema::DirectTootWasTooted.field_options
```

Comment:
Is the `subscription_scope` being inherited here? It seems like we're checking the value from the _same class_ where we manually configured it. 

I'd like to check that the value is inherited by subclasses, for example: 

```suggestion
      # Make a subclass without an explicit configuration 
      direct_toot_subclass = Class.new(SubscriptionFieldSchema::DirectTootWasTooted)
      # Then check if the field options got the inherited value
      direct_toot_options = direct_toot_subclass.field_options
```

How does that sound?

Reply:
Ah I see, slight confusion on my part but will update with these changes along with a small change to one of the other tests.